### ISSUE TITLE
RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by odeal4ik** [welldone-software/why-did-you-render#168]
Hello, i implemented your thing to my project
And it goes fine.

"@babel/preset-react": "^7.9.4",
"react": "^16.13.1",
"react-dom": "^16.13.1",
"react-redux": "^7.2.0",
"@welldone-software/why-did-you-render": "^6.0.3",

When i use it with [hrm], after some changes in files with code, it reloads fine.
But, if i press f5, for example, my app crashes with error **RangeError: Maximum call stack size exceeded...**

`RangeError: Maximum call stack size exceeded
    at Object.apply (C:\Develop\auchan-ecom-food-front\.yarn\$$virtual\@welldone-software-why-did-you-render-virtual-0f42de02d0\0\cache\@welldone-software-why-did-you-render-npm-6.0.3-f9d43ade3b-2.zip\node_modules\@welldone-software\why-did-you-render\src\whyDidYouRender.js:236:22)
    at Object.apply (C:\Develop\auchan-ecom-food-front\.yarn\$$virtual\@welldone-software-why-did-you-render-virtual-0f42de02d0\0\cache\@welldone-software-why-did-you-render-npm-6.0.3-f9d43ade3b-2.zip\node_modules\@welldone-software\why-did-you-render\src\whyDidYouRender.js:257:40)
    at Object.apply (C:\Develop\auchan-ecom-food-front\.yarn\$$virtual\@welldone-software-why-did-you-render-virtual-0f42de02d0\0\cache\@welldone-software-why-did-you-render-npm-6.0.3-f9d43ade3b-2.zip\node_modules\@welldone-software\why-did-you-render\src\whyDidYouRender.js:257:40)
    at Object.apply (C:\Develop\auchan-ecom-food-front\.yarn\$$virtual\@welldone-software-why-did-you-render-virtual-0f42de02d0\0\cache\@welldone-software-why-did-you-render-npm-6.0.3-f9d43ade3b-2.zip\node_modules\@welldone-software\why-did-you-render\src\whyDidYouRender.js:257:40)
    at Object.apply (C:\Develop\auchan-ecom-food-front\.yarn\$$virtual\@welldone-software-why-did-you-render-virtual-0f42de02d0\0\cache\@welldone-software-why-did-you-render-npm-6.0.3-f9d43ade3b-2.zip\node_modules\@welldone-software\why-did-you-render\src\whyDidYouRender.js:257:40)
    at Object.apply (C:\Develop\auchan-ecom-food-front\.yarn\$$virtual\@welldone-software-why-did-you-render-virtual-0f42de02d0\0\cache\@welldone-software-why-did-you-render-npm-6.0.3-f9d43ade3b-2.zip\node_modules\@welldone-software\why-did-you-render\src\whyDidYouRender.js:257:40)
    at Object.apply (C:\Develop\auchan-ecom-food-front\.yarn\$$virtual\@welldone-software-why-did-you-render-virtual-0f42de02d0\0\cache\@welldone-software-why-did-you-render-npm-6.0.3-f9d43ade3b-2.zip\node_modules\@welldone-software\why-did-you-render\src\whyDidYouRender.js:257:40)
    at Object.apply (C:\Develop\auchan-ecom-food-front\.yarn\$$virtual\@welldone-software-why-did-you-render-virtual-0f42de02d0\0\cache\@welldone-software-why-did-you-render-npm-6.0.3-f9d43ade3b-2.zip\node_modules\@welldone-software\why-did-you-render\src\whyDidYouRender.js:257:40)
    at Object.apply (C:\Develop\auchan-ecom-food-front\.yarn\$$virtual\@welldone-software-why-did-you-render-virtual-0f42de02d0\0\cache\@welldone-software-why-did-you-render-npm-6.0.3-f9d43ade3b-2.zip\node_modules\@welldone-software\why-did-you-render\src\whyDidYouRender.js:257:40)
    at Object.apply (C:\Develop\auchan-ecom-food-front\.yarn\$$virtual\@welldone-software-why-did-you-render-virtual-0f42de02d0\0\cache\@welldone-software-why-did-you-render-npm-6.0.3-f9d43ade3b-2.zip\node_modules\@welldone-software\why-did-you-render\src\whyDidYouRender.js:257:40)
    at Object.apply (C:\Develop\auchan-ecom-food-front\.yarn\$$virtual\@welldone-software-why-did-you-render-virtual-0f42de02d0\0\cache\@welldone-software-why-did-you-render-npm-6.0.3-f9d43ade3b-2.zip\node_modules\@welldone-software\why-did-you-render\src\whyDidYouRender.js:257:40)
    at Object.apply (C:\Develop\auchan-ecom-food-front\.yarn\$$virtual\@welldone-software-why-did-you-render-virtual-0f42de02d0\0\cache\@welldone-software-why-did-you-render-npm-6.0.3-f9d43ade3b-2.zip\node_modules\@welldone-software\why-did-you-render\src\whyDidYouRender.js:257:40)
    at Object.apply (C:\Develop\auchan-ecom-food-front\.yarn\$$virtual\@welldone-software-why-did-you-render-virtual-0f42de02d0\0\cache\@welldone-software-why-did-you-render-npm-6.0.3-f9d43ade3b-2.zip\node_modules\@welldone-software\why-did-you-render\src\whyDidYouRender.js:257:40)
    at Object.apply (C:\Develop\auchan-ecom-food-front\.yarn\$$virtual\@welldone-software-why-did-you-render-virtual-0f42de02d0\0\cache\@welldone-software-why-did-you-render-npm-6.0.3-f9d43ade3b-2.zip\node_modules\@welldone-software\why-did-you-render\src\whyDidYouRender.js:257:40)
    at Object.apply (C:\Develop\auchan-ecom-food-front\.yarn\$$virtual\@welldone-software-why-did-you-render-virtual-0f42de02d0\0\cache\@welldone-software-why-did-you-render-npm-6.0.3-f9d43ade3b-2.zip\node_modules\@welldone-software\why-did-you-render\src\whyDidYouRender.js:257:40)
    at Object.React.createElement (C:\Develop\auchan-ecom-food-front\.yarn\$$virtual\@welldone-software-why-did-you-render-virtual-0f42de02d0\0\cache\@welldone-software-why-did-you-render-npm-6.0.3-f9d43ade3b-2.zip\node_modules\@welldone-software\why-did-you-render\src\whyDidYouRender.js:257:40)`
    
here is config and import 
![image](https://user-images.githubusercontent.com/8823008/102994386-08019400-4530-11eb-9a0e-8bead7457d6c.png)    
![image](https://user-images.githubusercontent.com/8823008/102994269-ce308d80-452f-11eb-936c-e942bdd41e79.png)

i can't understand what to do.
if needed some more info, pls report

**Comment by odeal4ik** [comment]
Upd. tried to watch one component by option `include` , and got the same thing.

**Comment by vzaidman** [comment]
thanks ill look at it soon

**Comment by vzaidman** [comment]
can you please check what is on line 257 of:
`node_modules\@welldone-software\why-did-you-render\src\whyDidYouRender.js`
and also line 236

also, there is actually no "src" in `node_modules\@welldone-software\why-did-you-render` what kind of system do you use that it exists there?

**Comment by vzaidman** [comment]
also, what `webpack` / `create-react-app` do you use?

**Comment by odeal4ik** [comment]
> @odeal4ik
> can you please check what is on line 257 of:
> `node_modules\@welldone-software\why-did-you-render\src\whyDidYouRender.js`
> and also line 236
> 
> also, there is actually no "src" in `node_modules\@welldone-software\why-did-you-render` what kind of system do you use that it exists there?

I thought this is it ?!...
![image](https://user-images.githubusercontent.com/8823008/103036338-926ee580-457a-11eb-956a-2c9ac89d88b5.png)

win10

**Comment by odeal4ik** [comment]
> also, what `webpack` / `create-react-app` do you use?

![image](https://user-images.githubusercontent.com/8823008/103036431-ce09af80-457a-11eb-8441-5e1508c34caf.png)

**Comment by vzaidman** [comment]
how did you install the package?
```
npm install @welldone-software/why-did-you-render
```
?
because it has no `src`. only `dist`

**Comment by vzaidman** [comment]
also-
can you try adding this at the bottom of the file?
```
if (process.env.NODE_ENV === 'development') {
  if (module.hot) {
    module.hot.decline();
  }
}
```

**Comment by odeal4ik** [comment]
> how did you install the package?

by yarn yarn add @welldone-software/why-did-you-render --dev

**Comment by vzaidman** [comment]
try clearing cacher and reinstall
```
yarn cache clean
```
and
```
yarn install --force
```

**Comment by odeal4ik** [comment]
i use 
rm -rf .yarn/cache
rm -rf yarn.lock

then yarn install my dependencies, but now i have some problem with start my app, it lil crashes.

`
if (process.env.NODE_ENV === 'development') {
  if (module.hot) {
    module.hot.decline();
  }
}
`
you just want to "disable" hot in this mode?

**Comment by vzaidman** [comment]
only on the `wdyr` file

**Comment by odeal4ik** [comment]
for proof it module, and module.hot in wdyr.ts file, and problem is the same, it still crashes after page reload 
![image](https://user-images.githubusercontent.com/8823008/103076019-d7783380-45dd-11eb-96b9-baf7925c82a5.png)

![image](https://user-images.githubusercontent.com/8823008/103076282-6c7b2c80-45de-11eb-9367-26abb7bff79d.png)

**Comment by vzaidman** [comment]
are you using Yarn 2?

**Comment by odeal4ik** [comment]
oh yeah

**Comment by vzaidman** [comment]
Maybe try yarn 1? I dont know how it works

**Comment by odeal4ik** [comment]
oh, too many things were made for using yarn 2...i'll try on this project somehow to change on v1, but it seems to me, that i'll see more bugs with this.

**Comment by odeal4ik** [comment]
i tried on cra with yarn 1 and hot reload, and all goes fine (

**Comment by vzaidman** [comment]
i think it has to do with yarn2 cache. it looks bugged to me because there's no 
```
@welldone-software\why-did-you-render\src
```
only
```
@welldone-software\why-did-you-render\dist
```

I'm glad it works!

**Comment by odeal4ik** [comment]
Ok, i think it's a yarn problem, but, i patch my yarn config, and duplicate dependencies in node_modules, and they are there, it 100%.
![image](https://user-images.githubusercontent.com/8823008/103092330-f095da00-4607-11eb-9ebc-8d42b8f651e0.png)

But error still exist
![image](https://user-images.githubusercontent.com/8823008/103092255-b6c4d380-4607-11eb-901d-5deb438b232d.png)

And why it links to sourse libs files? Not to modules?
![image](https://user-images.githubusercontent.com/8823008/103092340-f7245180-4607-11eb-9549-367429e0392a.png)

**Comment by davidgilbertson** [comment]
I'm getting the same issue. I don't use yarn. Nothing happens on a few pages, then one page does this:
```
 RangeError: Maximum call stack size exceeded
[0]     at getWDYRType (C:\Users\david\web\keplar\node_modules\@welldone-software\why-did-you-render\dist\whyDidYouRender.js:1418:25)
[0]     at Object.React.createElement (C:\Users\david\web\keplar\node_modules\@welldone-software\why-did-you-render\dist\whyDidYouRender.js:1444:22)
[0]     at Object.React.createElement (C:\Users\david\web\keplar\node_modules\@welldone-software\why-did-you-render\dist\whyDidYouRender.js:1471:42)
[0]     at Object.React.createElement (C:\Users\david\web\keplar\node_modules\@welldone-software\why-did-you-render\dist\whyDidYouRender.js:1471:42)
[0]     at Object.React.createElement (C:\Users\david\web\keplar\node_modules\@welldone-software\why-did-you-render\dist\whyDidYouRender.js:1471:42)
[0]     at Object.React.createElement (C:\Users\david\web\keplar\node_modules\@welldone-software\why-did-you-render\dist\whyDidYouRender.js:1471:42)
```

I'm using Next.js which will be doing some server-rendering.

Setup is a copy/paste from the readme (I'm using TypeScript):
```ts
/// <reference types="@welldone-software/why-did-you-render" />
import React from 'react';

if (process.env.NODE_ENV === 'development') {
  const whyDidYouRender = require('@welldone-software/why-did-you-render');
  whyDidYouRender(React, {
    trackAllPureComponents: true,
  });
  // Tried with and without this:
  if (module.hot) {
    module.hot.decline();
  }
}
```

Node: `14.14.0`
npm: `6.14.0`

Some other versions, not sure what might be relevant
```json
    "@welldone-software/why-did-you-render": "^6.0.4",
    "next": "^9.5.5",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "typescript": "^4.1.3",
```

**Comment by vzaidman** [comment]
@davidgilbertson what version of Next.js / React?

**Comment by vzaidman** [comment]
ok! I've managed to reproduce the bug in next.js and fixed it in version [6.0.5]*https://github.com/welldone-software/why-did-you-render/releases/tag/v6.0.5)!

**Comment by davidgilbertson** [comment]
You work fast! 👏
